alf/nest/nest.py

```diff
@@ -691,3 +691,39 @@ def _traverse(nested, levels):
                             "a dict or namedtuple!")
 
     return _traverse(nested=nested, levels=field.split('.') if field else [])
+
+
+def set_field(nested, field, new_value):
+    """Set the field in nested to ``new_value``.
+
+    field is a string separated by ".". set_filed(nested, "a.b", v) is equivalent
+    to ``nested._replace(a=nested.a._replace(b=v))`` if nested is constructed
+    using namedtuple.
+
+    Args:
+        nested (nest): a nested structure
+        field (str): indicate the path to the field with '.' separating the field
+            name at different level
+        new_value (any): the new value for the field
+    Returns:
+        nest: a nest same as ``nested`` except the filed ``field`` replaced by
+            ``new_value``
+    """
+
+    def _traverse(nested, levels):
+        if not levels:
+            return new_value
+        level = levels[0]
+        if is_namedtuple(nested):
+            new_val = _traverse(
+                nested=getattr(nested, level), levels=levels[1:])
+            return nested._replace(**{level: new_val})
+        elif isinstance(nested, dict):
+            new_val = nested.copy()
+            new_val[level] = _traverse(nested=nested[level], levels=levels[1:])
+            return new_val
+        else:
+            raise TypeError("If value is a nest, it must be either " +
+                            "a dict or namedtuple!")
+
+    return _traverse(nested=nested, levels=field.split('.') if field else [])
```

Comment:
Can this be done using transform_nest, e.g.
`result = alf.nest.utils.transform_nest(result, field, lambda _: new_value)`?

Reply:
Sure it can. But not feeling well to use more complex API to do the simpler task.

Reply:
Right. But I think that to reduce maintenance cost, this can just be a wrapper instead that calls transform_nest

Reply:
If there isn't much efficiency gain, then I'd vote for less code.  Having set_field func call transform_nest is probably a good compromise.